### PR TITLE
backport: bitcoin#20459, #20842, #21557, #21840, #21867, #21897, #21900, #21945, #22048, #22057

### PR DESCRIPTION
**Pull request by knst**
## What was done?
Backports from v22 bitcoin.
Mostly related to MiniWallet and RPC improvements, see commits

## How Has This Been Tested?
Run unit/functional tests

## Breaking Changes
N/A

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
Mostly ok, one small issue

**Review by UdjinM6 — APPROVED** [review]
utACK b73f48f3b900ef88630c95ec5ed4a9a5ca2d6478

File: src/qt/test/rpcnestedtests.cpp [hunk 1]
```diff
@@ -11,8 +11,6 @@
 #include <univalue.h>
 #include <util/system.h>
 
-#include <llmq/context.h>
-
```
Comment (PastaPastaPasta) [inline src/qt/test/rpcnestedtests.cpp]: nit: 21840 unrelated?

Comment (knst) [reply src/qt/test/rpcnestedtests.cpp]: That's missing changes from https://github.com/dashpay/dash/pull/6112 - caught one more case but 6112 is merged already

Comment (PastaPastaPasta) [reply src/qt/test/rpcnestedtests.cpp]: would've preferred in separate PR, but this is fine

**Pull request merged**